tlair/src/test/scala/at/forsyte/apalache/tla/lir/transformations/standard/TestReTLALangaugePred.scala

```diff
@@ -0,0 +1,159 @@
+    expectOk(pred.isExprOk(appFun(name("f"), int(3))))
+    expectOk(pred.isExprOk(appFun(name("f"), tuple(int(3), int(3)))))
+    expectOk(pred.isExprOk(except(name("f"), tuple(int(3)), bool(true))))
+    expectOk(pred.isExprOk(except(name("f"), tuple(int(3), int(3)), bool(true))))
+    expectFail(pred.isExprOk(funSet(name("X"), name("Y"))))
+    expectFail(pred.isExprOk(dom(name("X"))))
+  }
+
+  test("reTLA: tuples, records, sequences") {
+    expectFail(pred.isExprOk(tuple(int(1), bool(true))))
+    expectFail(pred.isExprOk(enumFun(str("a"), bool(true))))
+    expectFail(pred.isExprOk(head(tuple(int(1), int(2)))))
+    expectFail(pred.isExprOk(tail(tuple(int(1), int(2)))))
+    expectFail(pred.isExprOk(subseq(tuple(int(1), int(2)), int(3), int(4))))
+    expectFail(pred.isExprOk(len(tuple(int(1), int(2)))))
+    expectFail(pred.isExprOk(append(tuple(int(1), int(2)), tuple(int(2), int(3)))))
+  }
+
+  test("reTLA: misc.") {
+    expectFail(pred.isExprOk(label(int(2), "a")))
+    expectFail(pred.isExprOk(label(int(2), "a", "b")))
+  }
+
+  test("reTLA: TLC support") {
+    expectFail(pred.isExprOk(OperEx(TlcOper.print, tla.bool(true), tla.str("msg"))))
+    expectFail(pred.isExprOk(OperEx(TlcOper.printT, tla.str("msg"))))
+    expectFail(pred.isExprOk(OperEx(TlcOper.assert, tla.bool(true), tla.str("msg"))))
+    expectFail(pred.isExprOk(OperEx(TlcOper.colonGreater, tla.int(1), tla.int(2))))
+    expectFail(pred.isExprOk(OperEx(TlcOper.atat, NameEx("fun"), NameEx("pair"))))
+  }
+
+  test("reTLA: integer expressions") {
+    expectFail(pred.isExprOk(lt(int(1), int(3))))
+    expectFail(pred.isExprOk(le(int(1), int(3))))
+    expectFail(pred.isExprOk(gt(int(1), int(3))))
+    expectFail(pred.isExprOk(ge(int(1), int(3))))
+    expectFail(pred.isExprOk(plus(int(1), int(3))))
+    expectFail(pred.isExprOk(minus(int(1), int(3))))
+    expectFail(pred.isExprOk(mult(int(1), int(3))))
+    expectFail(pred.isExprOk(div(int(1), int(3))))
+    expectFail(pred.isExprOk(mod(int(1), int(3))))
+    expectFail(pred.isExprOk(exp(int(2), int(3))))
+    expectFail(pred.isExprOk(uminus(int(2))))
+  }
+
+  test("reTLA: control expressions") {
+    expectOk(pred.isExprOk(ite(name("p"), int(1), int(2))))
+    expectFail(pred.isExprOk(caseOther(int(1), bool(false), int(2))))
+    expectFail(pred.isExprOk(caseSplit(bool(false), int(2))))
+  }
+
+  test("reTLA: record expressions") {
+    expectFail(pred.isExprOk(recSet(name("a"), name("A"))))
+  }
+
+  test("reTLA: logic expressions") {
+    expectOk(pred.isExprOk(or(bool(false), name("b"))))
+    expectOk(pred.isExprOk(and(bool(false), name("b"))))
+    expectOk(pred.isExprOk(equiv(bool(false), name("b"))))
+    expectOk(pred.isExprOk(impl(bool(false), name("b"))))
+    expectOk(pred.isExprOk(neql(bool(false), name("b"))))
+  }
+
+  /**
+   * **************************** the tests from TestFlatLanguagePred ********************************************
```

Comment:
I don't think I understand this comment. Why are there tests for a different predicate in this test suite? Does that mean we're running the same tests twice?

Reply:
They're unique tests for `pred.isExprOk`, just that this part tests for the presence of LET-operators _like_ the tests in `TestFlatLanguagePred`. I copied the comment with the rest and didn't bother changing it, but the tests are different.

Reply:
The fixed comment helps clear this up for me. Thanks!